### DISCUSSION TITLE
[question] Does electron-trpc support communication between windows?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [jsonnull/electron-trpc#215] [Q&A] [answered]
Hello:
Does electron-trpc support communication between windows?

As we know, usually a simple project has only one mainWindow. If other windows are created, can electron-trpc be used for communication?

**Comment by jsonnull** [comment, marked as answer]
It's possible to do this with vanilla Electron IPC like this:

- subscribe to an IPC channel from each window
- publish messages from one window to the main thread
- have the main thread broadcast messages to all windows

I can imagine you could adapt this to a tRPC pattern:

- create subscriptions in each window which wants to receive messages
- each subscription in the router is now part of a PubSub store of some kind
- send messages to the router from one window in a mutation and have the router handle the mutation by forwarding the message to the PubSub store which can then broadcast to active subscriptions
